buildSrc/src/main/java/gradlebuild/JniNokeePlugin.java

```diff
@@ -202,10 +404,7 @@ void createToolChains(NativeToolChainRegistry toolChainRegistry) {
             });
             toolChainRegistry.create("clang", Clang.class, toolChain -> {
                 // The core Gradle toolchain for Clang only targets x86 and x86_64 out of the box.
-                OperatingSystem os = new DefaultNativePlatform("current").getOperatingSystem();
-                if (os.isMacOsX()) {
-                    toolChain.target("macosaarch64");
-                }
+                toolChain.target("macosaarch64");
```

Comment:
❌ There is still some work to do here to avoid building aarch64 on x64 architecture.